books/spiders/spider.py

```diff
@@ -0,0 +1,62 @@
+import requests
+import scrapy
+from bs4 import BeautifulSoup
+from scrapy.http import Response
+
+
+class SpiderSpider(scrapy.Spider):
+    name = "spider"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com"]
+
+    def _parse_single_book(self, url: str) -> dict:
+        page = requests.get(url).content
```

Comment:
Using `requests` to fetch pages within a Scrapy spider is not recommended. Scrapy's own request and response handling should be used to take advantage of its asynchronous capabilities and settings like download delays and concurrency.